include/envoy/router/route_config_update_receiver.h

```diff
@@ -52,6 +52,11 @@ class RouteConfigUpdateReceiver {
    */
   virtual const std::string& configVersion() PURE;
 
+  /**
+   * @return bool return whether VHDS configuration has been changed in the last RDS update.
+   */
+  virtual bool vhdsConfigurationChanged() const PURE;
```

Comment:
How stable is this? What is multiple RDS updates occur? Will the VHDS change status persist somehow?

Reply:
@htuch, @lambdai: both of you have reservations re:  stability/safety of the state exposed by vhdsConfigurationChanged() and aliasesInLastVhdsUpdate() methods. 

I'm guessing your concern is with the internal state being accessed while another thread is updating it. RouteConfigUpdateReceiver instance is only meant to be accessed from onUpdate callbacks, and its state is consistent during that time: the state is updated at the start of a callbacks and then used until the callback is completed. The only exception to this is RouteConfigProviderManagerImpl::dump() that accesses lastUpdated timestamp (but from the same thread as Rds/Vhds callbacks do, the state still is consistent).

Perhaps a better way to factor this would be to split RouteConfigUpdateReceiver into a RouteConfig state and a last update state. The latter could be passed to callbacks as a parameter, which would make the intent and the lifecycle less muddled. I can add a todo for this?

Reply:
`the state is updated at the start of a callbacks and then used until the callback is completed.`
Got it.
IMHO it's more like stack variable. If it's not very easy to switch to stack variable you can also add the comment of the life time in the member declaration.